internal/commands/build.go

```diff
@@ -72,18 +72,30 @@ func Build(logger logging.Logger, cfg config.Config, packClient PackClient) *cob
 				logger.Debugf("Using project descriptor located at %s", style.Symbol(actualDescriptorPath))
 			}
 
+			builder := flags.Builder
```

Comment:
This ended up being a bit weird to implement than I imagined. 

We needed to do 2 things - 
1. Figure out if the user explicitly passed a flag or the default one was used.
2. Parse the project descriptor and check if builder existed before before we could validate the bulder flag. In order to do so, I moved the builder validation out of validate flags as it was sort of a chicken egg problem where I had to use the project descriptor from the flags and parse it before I could validate the actual flags and that seemed to break the whole contract of validating the flags before using them. So for now I extracted this part out of validate flags and I explicitly check and throw an error if builder is null.

Reply:
I also had to parse the builder and decide which one to use in the command instead of the client.Build as at that point it was too late to distinguish which flag was passed and the order of preference.